rotkehlchen/tasks/manager.py

```diff
@@ -269,61 +274,72 @@ def _maybe_schedule_xpub_derivation(self) -> Optional[list[gevent.Greenlet]]:
             ))
         return greenlets
 
-    def _maybe_query_ethereum_transactions(self) -> Optional[list[gevent.Greenlet]]:
-        """Schedules the ethereum transaction query task if enough time has passed"""
-        with self.database.conn.read_ctx() as cursor:
-            accounts = self.database.get_blockchain_accounts(cursor).eth
-            if len(accounts) == 0:
-                return None
-
-            now = ts_now()
-            dbevmtx = DBEvmTx(self.database)
-            queriable_accounts = []
-            for account in accounts:
-                _, end_ts = dbevmtx.get_queried_range(cursor, account, SupportedBlockchain.ETHEREUM)  # noqa: E501
-                if now - max(self.last_eth_tx_query_ts[account], end_ts) > ETH_TX_QUERY_FREQUENCY:
-                    queriable_accounts.append(account)
-
-        if len(queriable_accounts) == 0:
-            return None
+    def _maybe_query_evm_transactions(self) -> Optional[list[gevent.Greenlet]]:
+        """Schedules the evm transaction query task if enough time has passed"""
+        shuffled_chains = list(EVM_CHAINS_WITH_TRANSACTIONS)
+        random.shuffle(shuffled_chains)
+        for blockchain in shuffled_chains:
```

Comment:
I don't think you understood what I asked here. Here and below I asked to do it only for 1 chain, randomly each time. First chain that has something. Right now it seems it still happens for all. But you just randomized the order